mne/decoding/time_gen.py

```diff
@@ -387,40 +388,88 @@ def _predict_time_loop(X, estimators, cv, slices, predict_mode):
     """
     n_epochs = len(X)
     # Loop across testing slices
-    y_pred = [list() for _ in range(len(slices))]
 
-    # XXX EHN: This loop should be parallelized in a similar way to fit()
+    if predict_mode == 'cross-validation':
```

Comment:
This can be put down below, to avoid testing predict_mode several times, and merge these tests. See https://github.com/kingjr/mne-python/tree/gat_optimize2 for a working example if you want.